doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -90,11 +93,63 @@ Rewriting with Leibniz and setoid equality
      any of them can be rewritten.  If not specified, only the first occurrence
      in the conclusion is replaced.
 
-     If :n:`at @occs_nums` is specified, rewriting is always done with
-     :ref:`setoid rewriting <generalizedrewriting>`, even for Leibniz’s equality,
-     which means that you must `Require Setoid` to use that form.
-     However, note that :tacn:`rewrite` (even when using setoid rewriting) and
-     :tacn:`setoid_rewrite` don't behave identically (as already mentioned above).
+     .. note::
+
+        If :n:`at @occs_nums` is specified, rewriting is always done
+        with :ref:`setoid rewriting <generalizedrewriting>`, even for
+        Leibniz equality, which means that you must `Require
+        Setoid` to use that form.  However, note that :tacn:`rewrite`
+        (even when using setoid rewriting) and :tacn:`setoid_rewrite`
+        don't behave identically (as already mentioned above).
+
+     .. note::
+
+        For each selected hypothesis (and the conclusion),
+        :tacn:`rewrite` will instantiate the rewriting lemma first
+        before taking the `at` clause into account, which limits the
+        number of occurrences that can be rewritten.  You can force
+        :tacn:`rewrite` to instantiate the rewriting lemma differently
+        by using a `with` clause (see :n:`@one_term_with_bindings`) or
+        you can use :tacn:`setoid_rewrite`, which considers each
+        occurrence separately.
+
+        .. example::
+
+           .. coqtop:: none
+
+              Require Import Arith.
+
+           If we want to rewrite the right-hand side, it will not work
+           like this:
+
+           .. coqtop:: out
+
+              Lemma add_assoc x y : x + y = y + x.
+
+           .. coqtop:: all fail abort
+
+              rewrite Nat.add_comm at 2.
+
+           A solution is to explicitly provide a different
+           instantiation:
+
+           .. coqtop:: out
+
+              Lemma add_assoc x y : x + y = y + x.
```

Comment:
You think it's necessary to show this again given that it already appears only a few lines above?

Reply:
I did several tests and it looked better like this.

Reply:
For the record, the name doesn't actually show in the output. But I'll replace this in the code anyway.

My reason for switching from an example based on `add_assoc` to this one is that this one is clearer on which occurrence is the first one and which is the second one.